src/Features/Core/Portable/PullMemberUp/QuickAction/ClassPullerWithQuickAction.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp.QuickAction
+{
+    internal class ClassPullerWithQuickAction : AbstractMemberPullerWithQuickAction
+    {
+        internal readonly static ClassPullerWithQuickAction Instance = new ClassPullerWithQuickAction();
+
+        private ClassPullerWithQuickAction()
+        {
+        }
+
+        protected override bool IsSelectedMemberDeclarationAlreadyInDestination(INamedTypeSymbol destination, ISymbol selectedMember)
+        {
+            if (selectedMember is IFieldSymbol fieldSymbol)
+            {
+                // If there is a field with same name in destination, pull the selected field will cause error,
+                // so don't provide refactoring under this scenario
+                return destination.GetMembers(fieldSymbol.Name).Any(member => member.IsKind(SymbolKind.Field));
```

Comment:
won't this be true if there's *any* member with the name of that field in teh destination?  For example, if you have a property or method with that name, won't that fail as well?

Reply:
don't you need to do a similar check for properties (and probably events), not just fields?

Reply:
OK, I just played with the syntax here. For this case, seems like field must have its unique name in the class otherwise it would case error. I will change the logic here to '
 if there is any member have the same name as the selected field, then don't provide the refactoring option'